### ISSUE TITLE
TODO

### ISSUE THREAD

**Issue by jonathanong** [mongodb-utils/mongodb-next#3]
Wrap:
- [x] MongoClient
- [x] DB
- [ ] Indexes
- [x] map-reduce
- [ ] capped collections 
- [ ] oplogs

**Comment by joshrtay** [comment]
still looking for help on this TODO? 

also, any plans for buffering commands to a collection until connection is successful?

**Comment by jonathanong** [comment]
yeah help would be great. i'm moving away from mongodb so i'm probably going to work on this next.

maybe it's not documented, but what i do is `await db.connect` before starting the server. problem solved.

**Comment by joshrtay** [comment]
i see.

just curious... what are you switching to? postgresql?

**Comment by jonathanong** [comment]
for now, yeah. looking into rethinkdb as well.

**Comment by haoxins** [comment]
mongodb node driver supports promise now
https://github.com/mongodb-utils/mongodb-next/issues/13
